lib/bloc/menubloc/menu_bloc.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:bloc/bloc.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:nutrilab/bloc/menubloc/menu_event.dart';
+import 'package:nutrilab/bloc/menubloc/menu_state.dart';
+import 'package:nutrilab/models/menuitemmodel.dart';
+
+class MenuBloc extends Bloc<MenuEvent, MenuState> {
+  final String userId;
+
+  MenuBloc(this.userId) : super(MenuInitial()) {
+    on<LoadMenuItems>(_onLoadMenuItems);
+  }
+
+  Future<void> _onLoadMenuItems(
+      LoadMenuItems event, Emitter<MenuState> emit) async {
+    emit(MenuLoading());
+
+    try {
+      DocumentSnapshot userSnapshot =
+          await FirebaseFirestore.instance.collection('users').doc(userId).get();
+      Map<String, dynamic> userData =
+          userSnapshot.data() as Map<String, dynamic>;
+
+      List<String> likedItems = List<String>.from(userData['likedItems'] ?? []);
+      List<String> cartItems = List<String>.from(userData['cartItems'] ?? []);
+
+      QuerySnapshot snapshot = await FirebaseFirestore.instance.collection('menuitems').get();
+      List<MenuItemModel> menuItems = snapshot.docs.map((doc) {
+        Map<String, dynamic> data = doc.data() as Map<String, dynamic>;
+        bool isLiked = likedItems.contains(doc.id);
+        bool isInCart = cartItems.contains(doc.id);
+
+        return MenuItemModel(
```

Comment:
Better to create mapper to handle this logic, for example:
`class MenuItemMapper {
MenuItemModel fbToDisplayModel (Map<String, dynamic> fbModel) {
return MenuItemModel(...)
}
}`

Reply:
Would it be possible to pass isLiked and isInCart to the mapper?

Reply:
Yes, `MenuItemModel fbToDisplayModel (Map<String, dynamic> fbModel, bool isLiked, bool isInCart)`